### ISSUE TITLE
Java <-> Ruby timestamp convert incorrect value in milli precision.

### ISSUE THREAD

**Issue by hiroyuki-sato** [embulk/embulk#999]
Follow-up: https://github.com/embulk/embulk-output-bigquery/issues/76

* expect value: `2015-07-13 00:00:00.100000 +00:00`
* actual value: `2015-07-13 00:00:00.000100 +00:00`

Ref: https://github.com/hiroyuki-sato/embulk-support/blob/master/output-bigquery_076_milli_sec/00README.md

Reproduce code.

```yaml
in:
  type: file
  parser:
    type: csv
    charset: UTF-8
    newline: LF
    delimiter: ','
    quote: '"'
    escape: '"'
    trim_if_not_quoted: false
    skip_header_lines: 1
    allow_extra_columns: false
    allow_optional_columns: false
    columns:
    - {name: date, type: timestamp, format: '%Y-%m-%d'}
    - {name: timestamp, type: timestamp, format: '%Y-%m-%d %H:%M:%S.%N'}
    - {name: 'null', type: string}
    - {name: long, type: long}
    - {name: string, type: string}
    - {name: double, type: double}
    - {name: boolean, type: boolean}
  path_prefix: example.csv
filters:
  - type: insert
    columns: 
    - timestamp_format:
  - type: ruby_proc
    columns:
    - name: timestamp_format
      skip_nil: false
      proc: |
        ->(_,record) do
          if t = record["timestamp"]
            t.strftime("%Y-%m-%d %H:%M:%S.%6N %:z")
          else
            nil
          end
        end
  - type: column
    columns:
    - { name: timestamp_format }
          
out: {type: stdout}
```

input data.

```
date,timestamp,"null",long,string,double,boolean
2015-07-13,2015-07-13 00:00:00.100000,,90,l6lTsvxd,903.4,true
2015-07-13,2015-07-13 00:00:00.100000,,91,XoALSEQg,394.5,true
2015-07-13,2015-07-13 00:00:00.100000,,92,0hgDRI_m,810.9,true
2015-07-13,2015-07-13 00:00:00.100000,,93,KjCRAc-A,477.4,true
2015-07-13,2015-07-13 00:00:00.100000,,94,fyQVGlT8,725.3,true
2015-07-13,2015-07-13 00:00:00.100000,,95,FpBYRPWK,316.6,true
2015-07-13,2015-07-13 00:00:00.100000,,96,9ikvnUqp,369.5,true
2015-07-13,2015-07-13 00:00:00.100000,,97,RRNYDAzK,506.5,true
2015-07-13,2015-07-13 00:00:00.100000,,90,l6lTsvxd,903.4,false
2015-07-13,2015-07-13 00:00:00.100000,,91,XoALSEQg,394.5,false
2015-07-13,2015-07-13 00:00:00.100000,,92,0hgDRI_m,810.9,false
2015-07-13,2015-07-13 00:00:00.100000,,93,KjCRAc-A,477.4,false
2015-07-13,2015-07-13 00:00:00.100000,,94,fyQVGlT8,725.3,false
2015-07-13,2015-07-13 00:00:00.100000,,95,FpBYRPWK,316.6,false
2015-07-13,2015-07-13 00:00:00.100000,,96,9ikvnUqp,369.5,false
2015-07-13,2015-07-13 00:00:00.100000,,97,RRNYDAzK,506.5,false
```

result 

```
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
2015-07-13 00:00:00.000100 +00:00
```

**Comment by dmikurube** [comment]
Thanks for reporting the issue!

(As chatted out of GitHub) It looks coming from our misunderstanding of JRuby's `RubyTime#setNSec`, or `nsec` in `RubyTime.newTime` represents 4 to 9 decimal places, not entire the fractional parts. It means that `rubyTime.setNSec(345678)` sets the fraction part like `.000345678`, not `.345678000`. That behavior is not documented, though...

Since it's a problem just when creating `RubyTime`, it affects only Ruby plugins, not Java plugins. Will fix this in the next release (v0.9.6).

The code causing this issue is at: https://github.com/embulk/embulk/blob/v0.9.5/embulk-core/src/main/ruby/embulk/schema.rb#L42

----

Submitted a pull request for Javadoc in JRuby, then. https://github.com/jruby/jruby/pull/5130

**Comment by dmikurube** [comment]
Confirmed #1000 fixes it. For @hiroyuki-sato's configuration above, it output:

```
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
2015-07-13 00:00:00.100000 +00:00
```
